grype/version/kb_contraint.go

```diff
@@ -37,14 +37,14 @@ func (c kbConstraint) supported(format Format) bool {
 }
 
 func (c kbConstraint) Satisfied(version *Version) (bool, error) {
-	if c.raw == "" && version != nil {
-		// an empty constraint is always satisfied
-		return true, nil
-	} else if version == nil {
-		if c.raw != "" {
-			// a non-empty constraint with no version given should always fail
-			return false, nil
+	if c.raw == "" {
+		// an empty constraint is never satisfied
+		return false, &NonFatalConstraintError{
+			constraint: c,
+			version:    version,
+			message:    "Unexpected data in DB: Empty raw version constraint.",
 		}
+	} else if version == nil {
 		return true, nil
 	}
```

Comment:
nit: Since you're returning in the previous `if` block, you don't need the `else` here, so you could express this check as a distinct thought. For example:

```suggestion
	}
	
	if version == nil {
		return true, nil
	}
```